python/ray/autoscaler/_private/autoscaler.py

```diff
@@ -605,6 +625,12 @@ def schedule_node_termination(
             aggregate=operator.add,
         )
         self.nodes_to_terminate.append(node_id)
+        # Scale down entire multi-host replica of node being deleted
+        tags = self.provider.node_tags(node_id)
+        if TAG_RAY_REPLICA_INDEX in tags:
+            replica_id = tags[TAG_RAY_REPLICA_INDEX]
+            if replica_id not in self.replicas_to_delete:
```

Comment:
May be we can directly `self.replicas_to_delete.add(replica_id)` without `if replica_id not in self.replicas_to_delete`.

Reply:
Done in [dd22594](https://github.com/ray-project/ray/pull/43470/commits/dd2259433a2822217caa41167cdb63a99ce6bcd5).